pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -3883,85 +3885,2699 @@ func Test_server_MultiInsert(t *testing.T) {
+									{
+										Field:       "vector dimension size",
+										Description: err.Error(),
+									},
+								},
+							},
+							&errdetails.ResourceInfo{
+								ResourceType: ngtResourceType + "/ngt.MultiInsert",
+								ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+							})
+						return err
+					}(),
+				},
+			}
+		}(),
+		func() test {
+			req := genMultiInsertReq(Float, vector.Gaussian, 100, f32VecDim)
+			vid := req.Requests[0].Vector.Id
+			for i := 0; i < len(req.Requests); i++ {
+				req.Requests[i].Vector.Vector = []float32{}
+			}
+
+			return test{
+				name: "Boundary Value Testing case 9.3: Fail to MultiInsert with all vector with empty insert vector",
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:    name,
+					ip:      ip,
+					svcCfg:  defaultF32SvcCfg,
+					svcOpts: defaultSvcOpts,
+				},
+				want: want{
+					err: func() error {
+						err := errors.ErrIncompatibleDimensionSize(0, intVecDim)
+						err = status.WrapWithInvalidArgument("MultiInsert API Incompatible Dimension Size detected",
+							err,
+							&errdetails.RequestInfo{
+								RequestId:   vid,
+								ServingData: errdetails.Serialize(req),
+							},
+							&errdetails.BadRequest{
+								FieldViolations: []*errdetails.BadRequestFieldViolation{
+									{
+										Field:       "vector dimension size",
+										Description: err.Error(),
+									},
+								},
+							},
+							&errdetails.ResourceInfo{
+								ResourceType: ngtResourceType + "/ngt.MultiInsert",
+								ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+							})
+						return err
+					}(),
+				},
+			}
+		}(),
+		func() test {
+			req := genMultiInsertReq(Float, vector.Gaussian, 100, f32VecDim)
+			req.Requests[0].Vector.Id = req.Requests[1].Vector.Id
+
+			// set insert request config
+			cfg := &payload.Insert_Config{
+				SkipStrictExistCheck: false,
+			}
+			for i := 0; i < len(req.Requests); i++ {
+				req.Requests[i].Config = cfg
+			}
+
+			// set want
+			w := genObjectLocations(100, name, ip)
+			w.Locations[0].Uuid = req.Requests[0].Vector.Id
+
+			return test{
+				name: "Decision Table Testing case 1.1: Success to MultiInsert with 2 duplicated ID when SkipStrictExistCheck is false",
+				args: args{
+					ctx:  ctx,
+					reqs: req,
+				},
+				fields: fields{
+					name:    name,
+					ip:      ip,
+					svcCfg:  defaultF32SvcCfg,
+					svcOpts: defaultSvcOpts,
+				},
+				want: want{
+					wantRes: w,
+				},
+			}
+		}(),
+		func() test {
+			req := genMultiInsertReq(Float, vector.Gaussian, 100, f32VecDim)
+			for i := 1; i < len(req.Requests); i++ {
+				req.Requests[i].Vector.Id = req.Requests[0].Vector.Id
+			}
+
+			// set insert request config
+			cfg := &payload.Insert_Config{
+				SkipStrictExistCheck: false,
+			}
+			for i := 0; i < len(req.Requests); i++ {
+				req.Requests[i].Config = cfg
+			}
```

Comment:
It is because `Config` can be nil, by default I think it should be nil so I explicitly set it here.

Reply:
> It is because Config can be nil, by default I think it should be nil so I explicitly set it here.

It is not a problem.
For example, it works with nil input.
```go
genMultiInsertReq := func(t objectType, dist vector.Distribution, num int, dim int, cfg ...*payload.Insert_Config) *payload.Insert_MultiRequest
....
    for i:=0; i < num; i++ {
    ...
        if cfg != nil {
            req.Requests[i].Config = cfg[0]
        }
    }
...
}
```
Or, implementing functional option works.

Reply:
fixed here.

https://github.com/vdaas/vald/pull/1612/commits/4e32995ae28cb1597bb354f95d021482fe952df9